### PR TITLE
ARCv3:HS5x_haps: change RAM base to address 0

### PR DESCRIPTION
**Pull request by pavelvkozlov**
Since FPGA images are now built with DDR mapped to address 0 update the LINUX_RAM_BASE and LINUX_LINK_BASE in default configs for hs5x and create new haps_hs5x.dts and haps_hs5x_idu.dts.
To avoid duplications created extra skeleton_haps.dtsi and skeleton_haps_idu.dtsi to use in all haps dts.

### CODE REVIEW

**Review by abrodkin — COMMENTED** [review]
Looks good to me, thanks!

File: arch/arc/boot/dts/haps_hs5x.dts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+/include/ "skeleton_hs.dtsi"
+/include/ "skeleton_haps.dtsi"
+
+/ {
+	model = "snps,zebu_hs5x";
+	compatible = "snps,zebu_hs";
+
+	memory {
+		device_type = "memory";
```
Comment (evgeniy-paltsev) [inline arch/arc/boot/dts/haps_hs5x.dts]: Note that technically you don't need to have device type overwrite here - as it's same as in the entry you update.

File: arch/arc/boot/dts/haps_hs5x.dts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+		/* CONFIG_LINUX_RAM_BASE needs to match low mem start */
+		reg = <0x0 0x00000000 0x0 0x60000000>;	/* 1.5 GB low mem */
+	};
+
+	chosen {
+		bootargs = "earlycon=uart8250,mmio32,0xf0000000,115200n8 console=ttyS0,115200n8 debug print-fatal-signals=1";
+	};
```
Comment (evgeniy-paltsev) [inline arch/arc/boot/dts/haps_hs5x.dts:21]: You can move chosen to dtsi as well - as I can see it's have same value everywhere.

File: arch/arc/boot/dts/haps_hs5x.dts (hunk 1 above)
Comment (pavelvkozlov) [reply arch/arc/boot/dts/haps_hs5x.dts]: It makes sense. I see arm .dts also drop this. I will drop it.

File: arch/arc/boot/dts/haps_hs5x.dts (hunk 2 above)
Comment (pavelvkozlov) [reply arch/arc/boot/dts/haps_hs5x.dts:21]: I agree, it is some sort of duplication, but I thought it's not bad to have bootargs near in dts file as it is something we could want to change often during debug and bring up. Also axs103.dts and axs101.dts contains bootargs in dts.

**Comment by pavelvkozlov** [comment]
Thank all for review. I'm going to merge.

**Pull request merged**